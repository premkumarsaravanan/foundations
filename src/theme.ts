export interface Theme {
  base: Base
  button: Button
  colors: Colors
}

export interface Base {
  font: Font
}

export interface Font {
  family: string
  sizes: Sizes
}

export interface Sizes {
  base: string
  headings: Headings
}

export interface Headings {
  h1: string
  h2: string
  h3: string
  h4: string
  h5: string
}

export interface Button {
  background: string
  color: string
}

export interface Colors {
  primary: string
  secondary: string
  icon: string
  base: string
  widgetHeading: string
  background: string
  warning: string
  loading: string
}

export const theme: Theme = {
  base: {
    font: {
      family: "'Open Sans', sans-serif",
      sizes: {
        base: '1.5rem',
        headings: {
          h1: '2.4rem',
          h2: '2.2rem',
          h3: '2.0rem',
          h4: '1.8rem',
          h5: '1.6rem'
        }
      }
    }
  },
  button: {
    background: '#ffffff',
    color: '#12263f'
  },
  colors: {
    primary: '#00a569',
    secondary: '#828085',
    icon: 'gray',
    base: '#12263f',
    widgetHeading: '#ffffff',
    background: '#ffffff',
    warning: '#d3033d',
    loading: '#95aac9'
  }
}
