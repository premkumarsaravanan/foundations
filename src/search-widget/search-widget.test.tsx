import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import SearchWidget from './search-widget'
import { theme } from '../theme'

describe('Search widget', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<SearchWidget theme={theme} API_KEY="" />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
  it('should match snapshot', () => {
    const wrapper = shallow(<SearchWidget theme={theme} API_KEY="" />)
    expect(wrapper).toMatchSnapshot()
  })
})
