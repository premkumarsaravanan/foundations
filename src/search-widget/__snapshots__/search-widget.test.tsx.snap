// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search widget should match snapshot 1`] = `
<ThemeProvider
  theme={
    Object {
      "base": Object {
        "font": Object {
          "family": "'Open Sans', sans-serif",
          "sizes": Object {
            "base": "1.5rem",
            "headings": Object {
              "h1": "3rem",
              "h2": "2.2rem",
              "h3": "2.0rem",
              "h4": "1.8rem",
              "h5": "1.6rem",
            },
          },
        },
      },
      "button": Object {
        "background": "#887d97",
        "color": "#ffffff",
      },
      "colors": Object {
        "background": "#ffffff",
        "base": "#12263f",
        "icon": "gray",
        "loading": "#95aac9",
        "primary": "#887d97 ",
        "searchResult": "#7ca091",
        "secondary": "#828085",
        "warning": "#d3033d",
        "widgetHeading": "#ffffff",
      },
      "searchWidget": Object {
        "backgroundColor": "rgba(124, 160, 145, 0.7)",
      },
    }
  }
>
  <styled.div>
    <ContextProvider
      value={
        Object {
          "err": null,
          "getCountResult": [Function],
          "getErrorString": [Function],
          "getResultArr": [Function],
          "isLoading": false,
          "propertyImages": Object {},
          "result": undefined,
          "searchKeyWord": "",
          "searchType": undefined,
          "selectedProperty": undefined,
          "setFetchError": [Function],
          "setFetchResult": [Function],
          "setPropertyImages": [Function],
          "setSearchKeyWord": [Function],
          "setSelectedProperty": [Function],
          "setStartFetching": [Function],
          "theme": Object {
            "base": Object {
              "font": Object {
                "family": "'Open Sans', sans-serif",
                "sizes": Object {
                  "base": "1.5rem",
                  "headings": Object {
                    "h1": "3rem",
                    "h2": "2.2rem",
                    "h3": "2.0rem",
                    "h4": "1.8rem",
                    "h5": "1.6rem",
                  },
                },
              },
            },
            "button": Object {
              "background": "#887d97",
              "color": "#ffffff",
            },
            "colors": Object {
              "background": "#ffffff",
              "base": "#12263f",
              "icon": "gray",
              "loading": "#95aac9",
              "primary": "#887d97 ",
              "searchResult": "#7ca091",
              "secondary": "#828085",
              "warning": "#d3033d",
              "widgetHeading": "#ffffff",
            },
            "searchWidget": Object {
              "backgroundColor": "rgba(124, 160, 145, 0.7)",
            },
          },
        }
      }
    >
      <styled.div>
        <styled.div>
          Find your perfect home
        </styled.div>
        <styled.div>
          Search for a property from Agent and Sons
        </styled.div>
        <styled.div>
          <styled.input
            name="search"
            onChange={[Function]}
            placeholder="e.g Town or Postcode"
          />
          <styled.button
            disabled={false}
            onClick={[Function]}
          >
            FOR SALE
          </styled.button>
          <styled.button
            disabled={false}
            onClick={[Function]}
          >
            TO RENT
          </styled.button>
        </styled.div>
      </styled.div>
    </ContextProvider>
  </styled.div>
</ThemeProvider>
`;
